samcli/commands/local/cli_common/options.py

```diff
@@ -48,7 +48,26 @@ def local_common_options(f):
             default=False,
             help="If set, will emulate a shutdown event after the invoke completes, "
             "in order to test extension handling of shutdown behavior.",
-        )
+        ),
+        click.option(
+            "--container-host",
+            default="localhost",
+            show_default=True,
+            help="Host of locally emulated Lambda container. "
+            "This option is useful when the container runs on a different host than SAM CLI. "
+            "For example, if you want to run SAM CLI in a Docker container on macOS, "
+            "use this option with host.docker.internal",
+        ),
+        click.option(
+            "--container-host-interface",
+            default="127.0.0.1",
+            show_default=True,
+            help="Interface that Docker host binds ports to. "
+            "This option is useful when users want to bind ports to a different interface. "
+            "It will bind ports to loopback interface by default. "
+            "For example, using 0.0.0.0 will bind ports to all interfaces. "
+            "Note: the input should be an IP address. ",
```

Comment:
More succinct?
```suggestion
            help="IP address of the host network interface that container ports should bind to. "
            "Use 0.0.0.0 to bind to all interfaces."
```

Reply:
should we add a differentiation in the help between the two options?

Reply:
I think the name and help text are enough to explain the difference between the two options. And from comments of #2492, customers who needs to use these 2 options know clearly the difference. For now, adding additional differentiation seems a little verbose for me.